src/ssh/azext_ssh/target_properties_utils.py

```diff
@@ -12,72 +12,67 @@
 
 logger = log.get_logger(__name__)
 
-
 # Send target OS type telemetry and check if authentication options are valid for that OS.
-def handle_target_os_type(cmd, op_info):
+def handle_target_machine_properties(cmd, op_info):
```

Comment:
I think this is much better, but I still have some thoughts for simplifying this code (you might disagree with me, this is just some brainstorming). It's interesting to have each function have a single responsibility related to a task. That way, if something changes, you don't have to change multiple functions. 
For example, consider that in the future we decide that we want to retrieve another property from the target resources. To do that we will need to change the handle_target_machine_properties function in multiple places (line 24, 27, 29, and add a function call to handle that property), and we will have to add changes to each function that makes a call to azure. 
So, I'm thinking, maybe instead of having the _get_*_property parse the response from azure, we could make it so the functions that handle each result have enough intelligence to get the value that it wants from the response.
For example:

handle_target_machine_properties:
    -call _get_*_property()
    -call _check_valid_os_type()
    -call _check_valid_agent_version()
    -call _configure_port_from_tag()

_check_valid_os_type(az_response):
    - call _get_os_property_from_azure_response():
    - check os type, print warning if needed

_get_os_property_from_azure_response():
    - check the resource type, get os property

What do you think?